zebra-chain/src/block/height.rs

```diff
@@ -47,19 +50,48 @@ impl Height {
     pub const MAX_AS_U32: u32 = Self::MAX.0;
 }
 
+impl Add<Height> for Height {
+    type Output = Option<Height>;
+
+    fn add(self, rhs: Height) -> Option<Height> {
+        // We know that both values are positive integers. Therefore, the result is
+        // positive, and we can skip the conversions. The checked_add is required,
+        // because the result may overflow.
```

Comment:
We don't have to do it as part of this PR but we should probably change `Height` to be like `Amount` where we guarantee that heights can never be invalid. We'll need to make the inner value of the height private and create new `TryFrom` impls and probably fix a bunch of code that this breaks across zebra. Once we do tho we won't even need to use `checked_add` because `Height::MAX * 2` won't overflow.

Reply:
That's now #1113.